src/pybind/cephfs/cephfs.pyx

```diff
@@ -1110,7 +1110,7 @@ cdef class LibCephFS(object):
 
         with nogil:
             # FIXME: replace magic number with CEPH_STATX_BASIC_STATS
-            ret = ceph_statx(self.cluster, _path, &stx, 0x7ffu, 0)
+            ret = ceph_statx(self.cluster, _path, &stx, 0x1000u, 0)
```

Comment:
This mask looks wrong:

`#define CEPH_STATX_VERSION      0x00001000U     /* Want/got stx_version */`

...so you're just requesting the CEPH_STATX_VERSION here. You might get other stuff, or you might not. You probably want to select everything (CEPH_STATX_ALL_STATS). Also, like the comment above it says, it'd be good to express the request mask symbolically.

Also, is this really what you want to do? Requesting STATX_BASIC_STATS allows you to emulate a stat() call and that's what this function is doing. Would it not be better to leave the stat and fstat functions alone and instead add a new statx and fstatx that could take a CEPH_STATX_* mask parameter from the caller?

Reply:
@jtlayton thank you
Here are all stat information, so I have not separated them.
At the same time I did not see the trigger interface compatibility issue.

Reply:
I'm not a fan on this PR.

One of the nice things about statx and ceph is that you can specify exactly what information you're interested in, and that could mean less caps being revoked when you aren't interested in everything inside StatResult.

I think it would be best to leave the stat/fstat functions alone, and introduce new statx and fstatx routines. That would allow the caller to specify a request mask, and they could get the btime and version iff they were interested in them.

Alternately, maybe there's a way to just add a new argument to stat and fstat functions in here? Make it default to CEPH_STATX_BASIC_STATS and the legacy behavior would be preserved.

If you do decide to do this after all, then at the very least you'll need to fix the comment above the ceph_statx call since it'll be wrong after this change.